tests/system/Config/fixtures/.env

```diff
@@ -32,3 +32,8 @@ SimpleConfig.crew.captain = Malcolm
 SimpleConfig.crew.pilot = Wash
 SimpleConfig.crew.comms = true
 SimpleConfig.crew.doctor = false
+SimpleConfig.crew.ranger = null
+SimpleConfig.crew.poacher = empty
+
+SimpleConfig.shortie = '12.45'
+SimpleConfig.longie = '250'
```

Comment:
I think `'12.45'` or `"12.45"` is string, `12.45` is float.

Reply:
So you mean drop the trimming of quotes? And regard encapsed numbers as intended to be strings?

Reply:
As a PHP user, I think `'12.45'` is a string, not an integer.
It is quoted intentionally.

I think adding the following two cases are enough.
`SimpleConfig.shortie = 12.45` as float, and `SimpleConfig.longie = 250` as integer.

Reply:
However, inside a `.env` file where everything is string, quotes really serve no purpose other than capturing in-between spaces. That's why in `DotEnv::sanitizeValue()` we capture what's inside those quotes.
https://github.com/codeigniter4/CodeIgniter4/blob/968c0278a0ba4138cf3e64a2bbab5f3a800532d5/system/Config/DotEnv.php#L163

So, in the absence of a whitespace in between, `12.45 === '12.45'`
And in the default `env`, you can write strings with or without quotes. It doesn't matter as long there is no whitespace.
> # database.tests.password = root
> # contentsecuritypolicy.styleSrc = 'self'

Reply:
I think this is an opportunity to standardize, and I like the ability to retain numerical strings. I can't think off the top of my head where that might be important (maybe with strongly typed Config properties?) but my hunch is we need to support it and checking for the presence of quotes is a decent approach.

Reply:
The current code does not matter.
`12.45 === '12.45'` is wired.
And there is no way to set numerical strings.

Reply:
First of all, all environment variable's values are strings in PHP.
And we don't have to use DotEnv, we can set environment variables to web server.
So the DotEnv code does not matter.

Now we are going to change the interpretation of the values. We already changed when `true`/`false`.
We can decide the behavior for us.

Reply:
Yes, integer 12 and string 12 are different.
Not being able to use numeric strings for properties in the Config files is too big a restriction.

I just wanted to say, the following two are different things. They are related and must work with, but not one.
- DotEnv reads `.env` and sets environment variables in PHP
- BaseConfig fetches environment variables and sets properties